src/metabase/api/card.clj

```diff
@@ -258,8 +273,9 @@
 (defn- check-allowed-to-modify-query
   "If the query is being modified, check that we have data permissions to run the query."
   [card-before-updates card-updates]
-  (when (api/column-will-change? :dataset_query card-before-updates card-updates)
-    (check-data-permissions-for-query (:dataset_query card-updates))))
+  (let [card-updates (m/update-existing card-updates :dataset_query mbql.normalize/normalize)]
```

Comment:
Could use `when-let` here?

Reply:
I don't think `when-let` would work because we have to normalize `dataset_query` before we pass it to `api/column-will-change?`, and `api/column-will-change?` expects the full map rather than just a single value (it checks whether the key is present, because there's a different between passing in `{:dataset_query nil}` (set the query to nil) and passing in `{}` (don't change dataset query